### PR TITLE
fix(sdcm/): Create tester tests, and improve final event

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/ubYvHzrF/2095-test-reports-error-and-passed

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
Passed PR provision and https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-byo-longevity/326/execution/node/47/log/

File: sdcm/sct_events.py [hunk 1]
```diff
@@ -277,48 +277,42 @@ class TestResultEvent(SctEvent, Exception):
...
+        return result
 
     def __reduce__(self):
         """Needed to be able to serialize and deserialize this event via pickle
         """
-        return self.__class__, (self.test_errors, self.framework_errors)
+        return self.__class__, (self.test_status, self.events)
 
-    def __eq__(self, other):
+    def __eq__(self, other: 'TestResultEvent'):
```
Comment (bentsi) [inline sdcm/sct_events.py:311]: why a typing hint of `other` is `str`?

File: sdcm/send_email.py [hunk 2]
```diff
@@ -124,7 +124,7 @@ class BaseEmailReporter:
     email_template_file = 'results_base.html'
     last_events_limit = 5
     last_events_body_limit_per_severity = 3000
-    last_events_severities = ['CRITICAL', 'ERROR', 'WARNING']
+    last_events_severities = ['CRITICAL', 'ERROR', 'TEST ERRORS', 'FRAMEWORK ERRORS', 'WARNING']
```
Comment (bentsi) [inline sdcm/send_email.py:127]: `FRAMEWORK ERRORS`, `TEST ERRORS` is not a severity...

File: unit_tests/test_tester.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+class ClusterTesterForTests(ClusterTester):
+    _sct_log = None
+    _final_event = None
+    _get_event_summary_cached = None
+    _get_events_grouped_by_category_cached = None
+
+    def __init__(self, *args):
+        self.logdir = os.path.join('/tmp', generate_random_string(10))
+        os.mkdir(self.logdir)
+        logging.config.dictConfig({
```
Comment (bentsi) [inline unit_tests/test_tester.py:30]: generalize `sdcm.utils.log.configure_logging` and use it here instead of a copy-paste

**Pull request merged**